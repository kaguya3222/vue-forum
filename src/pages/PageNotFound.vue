<template>
  <div class="col-full col-full_centered">
    <h1 class="col-full__header col-full__header_size_l">
      Not found
    </h1>
    <p class="col-full__text col-full__text_size_l">
      Ooops! We couldn't find what you are looking for.
    </p>
    <p class="col-full__text col-full__text_size_l">
      <router-link :to="{ name: 'Home' }">Back to Home Page</router-link>
    </p>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss">
.default-text {
  color: #4e9c7f;
}

.col-full__header_size_l {
  font-size: 100px;
}

.col-full__text_size_l {
  font-size: 50px;
  font-weight: 100;
}

.col-full_centered {
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: center;
}
</style>
