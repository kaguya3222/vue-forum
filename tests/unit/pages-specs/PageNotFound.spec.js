import { mount } from "@vue/test-utils";
import PageNotFound from "../../../src/pages/PageNotFound";

describe("PageNotFound", () => {
  const wrapper = mount(PageNotFound, {
    stubs: {
      "router-link": true
    }
  });
  test("Shows not found error and instructions", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Shows 'Back to home' title as a link", () => {
    const backToHomeLink = wrapper.find("router-link-stub");
    expect(backToHomeLink.element.textContent).toBe("Back to Home Page");
  });
});
