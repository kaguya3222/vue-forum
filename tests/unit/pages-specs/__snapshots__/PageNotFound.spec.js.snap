// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound Shows not found error and instructions 1`] = `
<div class="col-full col-full_centered">
  <h1 class="col-full__header col-full__header_size_l">
    Not found
  </h1>
  <p class="col-full__text col-full__text_size_l">
    Ooops! We couldn't find what you are looking for.
  </p>
  <p class="col-full__text col-full__text_size_l">
    <router-link-stub to="[object Object]">Back to Home Page</router-link-stub>
  </p>
</div>
`;
